src/uthash.h

```diff
@@ -264,17 +295,59 @@ do {
 } while (0)
 #endif
 
-#define HASH_ADD_KEYPTR_BYHASHVALUE_INORDER(hh,head,keyptr,keylen_in,hashval,add,cmpfcn) \
+#if HASH_NOMEM_OK
+
+#define HASH_ADD_TO_TABLE(hh,head,keyptr,keylen_in,hashval,add,op,mem_ok)        \
```

Comment:
I think I like this. :)
Nit: `op` should be spelled `where`, for consistency with `HASH_FSCK`. But on line 319 I'm proposing you hoist the `HASH_FSCK` out of this function, at which point you don't need `op`/`where` as a parameter at all, I think.